metrics/report/report_dockerfile/tidy_scaling.R

```diff
@@ -0,0 +1,261 @@
+
+			########################################################
+			#### Now extract all the pod launch boot data items ####
+			########################################################
+			local_bootdata=tibble(launch_time=br$launch_time$Result)
+			local_bootdata=cbind(local_bootdata, n_pods=br$n_pods$Result)
+			local_bootdata=cbind(local_bootdata, node=lp$node)
+			local_bootdata=cbind(local_bootdata, testname=rep(testname, length(local_bootdata$node)))
+
+
+			########################################################
+			#### Now extract all node performance information ######
+			########################################################
+			nu=br$node_util
+
+			# We need to associate a pod count with each result, but you
+			# get one result per-node, and the JSON does not carry the pod
+			# count in that table. Walk the node util structure, assigning the
+			# n_pods value from the boot results over to the list of node util
+			# entries associated with it - creating a new 'n_pods' field in the
+			# node util dataframe.
+			for (n in seq(length(br$n_pods$Result))) {
+				nu[[n]]$n_pods = br$n_pods$Result[[n]]
+			}
+
+			# node_util is a list of nested data frames. I'm sure there is some better R'ish
+			# way of extracting this data maybe with dplyr, map, select or melt, but I can't
+			# work it out right now, and at least this is semi-readable...
+			#
+			# Basically, we are de-listing and flattening the lists of dataframes into a
+			# singly 'tidy' dataframe...
+			nodes=do.call("rbind", lapply(nu, "[", "node"))
+			noschedule=do.call("rbind", lapply(nu, "[", "noschedule"))
+			n_pods=do.call("rbind", lapply(nu, "[", "n_pods"))
+			idle=lapply(nu, "[", "cpu_idle")
+			idle_df=do.call("rbind", lapply(idle, "[[", "cpu_idle"))
+			free=lapply(nu, "[", "mem_free")
+			free_df=do.call("rbind", lapply(free, "[[", "mem_free"))
+			used=lapply(nu, "[", "mem_used")
+			used_df=do.call("rbind", lapply(used, "[[", "mem_used"))
+
+			# and build our rows
+			local_nodedata=tibble(node=nodes$node)
+			local_nodedata=cbind(local_nodedata, n_pods=n_pods)
+			local_nodedata=cbind(local_nodedata, noschedule=noschedule)
+			local_nodedata=cbind(local_nodedata, idle=idle_df$Result)
+			local_nodedata=cbind(local_nodedata, mem_free=free_df$Result)
+			local_nodedata=cbind(local_nodedata, mem_used=used_df$Result)
+			local_nodedata=cbind(local_nodedata, testname=rep(testname, length(local_nodedata$node)))
+
+			# Now Calculate some stats. This gets more complicated as we may have n-nodes,
+			# and we want to show a 'pod average', so we try to assess for all nodes. If
+			# we have different 'size' nodes in a cluster, that could throw out the result,
+			# but the only other option would be to try and show every node separately in the
+			# table.
+
+			# Get a list of all the nodes
+			nodes=unique(local_nodedata$node)
+
+			memtotal=0
+			cputotal=0
+			# Calculate per-node totals, and tot them up to a global total.
+			for (n in nodes) {
```

Comment:
There should be a bypass for CPU and memory if noschedule == true

Reply:
adding...